File: src/ECommerce.Domain/Entities/Product.cs
using ECommerce.Domain.Common;

namespace ECommerce.Domain.Entities;

public class Product : AuditableEntity
{
    public required string Name { get; set; }
    public string? Description { get; set; }
    public required string Slug { get; set; }
    public required string Sku { get; set; }
    public decimal Price { get; set; }
    public decimal? CompareAtPrice { get; set; }
    public decimal? CostPrice { get; set; }
    public int StockQuantity { get; set; }
    public int LowStockThreshold { get; set; } = 10;
    public bool IsActive { get; set; } = true;
    public bool IsFeatured { get; set; }
    public ICollection<ProductImage> Images { get; set; } = [];
    public ICollection<Tag> Tags { get; set; } = [];
    public double Weight { get; set; }
    public string? Dimensions { get; set; }
    public bool IsDeleted { get; set; } = false;

    // Foreign Keys
    public Guid CategoryId { get; set; }
    public Guid? BrandId { get; set; }

    // Navigation Properties
    public Category Category { get; set; } = null!;
    public Brand? Brand { get; set; }
    public ICollection<Review> Reviews { get; set; } = [];
    public ICollection<OrderItem> OrderItems { get; set; } = [];
}

File: cooksafari_app/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  slug: string;
  description?: string;
  sku: string;
  price: number;
  discountPrice?: number;
  stockQuantity: number;
  imageUrl?: string;
  images?: string[];
  categoryId: string;
  categoryName?: string;
  brandId?: string;
  brandName?: string;
  rating?: number;
  reviewCount?: number;
  isFeatured?: boolean;
}

export interface Category {
  id: string;
  name: string;
  slug: string;
  imageUrl?: string;
  productCount?: number;
}
